src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdApi.java

```diff
@@ -59,7 +59,31 @@ public CcdApi(
 
     private CaseDetails retrieveCase(String caseRef, String jurisdiction) {
         CcdAuthenticator authenticator = authenticatorFactory.createForJurisdiction(jurisdiction);
-        return feignCcdApi.getCase(authenticator.getUserToken(), authenticator.getServiceToken(), caseRef);
+        try {
+            return feignCcdApi
+                .getCase(authenticator.getUserToken(), authenticator.getServiceToken(), caseRef);
+        } catch (FeignException ex) {
+            removeFromIdamCache(ex, jurisdiction);
+            throw ex;
+        }
```

Comment:
Maybe it's a bit too much? If it's 502 or 404 or 422 etc., there's no need to remove token from cache.

Reply:
Maybe we should just restrict it to 401 and 403.

Reply:
it removes when it is 401 or 403. inside the method `removeFromIdamCache`

Reply:
What I mean is we're removing the token from the cache every time we get a 4xx or 5xx response, instead of just 401 and 403. Is there a need for it?

Reply:
Aaa, now I get it, sorry